rust/feature-flags/src/handler/flags.rs

```diff
@@ -10,29 +10,181 @@ use crate::{
     },
 };
 use axum::extract::State;
+use serde::{Deserialize, Serialize};
 use serde_json::Value;
 use std::collections::HashMap;
 use uuid::Uuid;
 
 use super::{evaluation, types::FeatureFlagEvaluationContext};
 use crate::router;
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum EvaluationRuntime {
+    All,
+    Client,
+    Server,
+}
+
+impl From<String> for EvaluationRuntime {
+    fn from(s: String) -> Self {
+        EvaluationRuntime::from(s.as_str())
+    }
+}
+
+impl From<&str> for EvaluationRuntime {
+    fn from(s: &str) -> Self {
+        if s.eq_ignore_ascii_case("client") {
+            EvaluationRuntime::Client
+        } else if s.eq_ignore_ascii_case("server") {
+            EvaluationRuntime::Server
+        } else {
+            EvaluationRuntime::All
+        }
+    }
+}
+
+/// Determines the evaluation runtime based on request characteristics.
+/// Uses explicit runtime if provided, otherwise analyzes user-agent, headers, and lib_version
+/// to detect if the request is from a client-side (browser/mobile) or server-side SDK.
+fn detect_evaluation_runtime_from_request(
```

Comment:
What is the expected interaction when someone passes a set of flags via `flags_to_evaluate` but some of those flags don't match the current runtime? Should we return the intersection? Or should `flags_to_evaluate` take priority. I think the latter makes sense because they were explicitly specified. We probably should have a test for it.

Reply:
I disagree and I think it should be the other way around.  `flags_to_evaluate` is mostly ever just one flag coming from server-side SDKs, and I don't think those SDKs should be able to trigger evaluation of client-side flags.

Reply:
I can live with that.